packages/connect-popup/src/log.tsx

```diff
@@ -0,0 +1,112 @@
+import React, { useEffect, useState } from 'react';
+import { createRoot } from 'react-dom/client';
+
+import { ObjectInspector } from 'react-inspector';
+
+const DownloadButton = ({ array, filename }: { array: any[]; filename: string }) => {
+    const buttonStyle = {
+        backgroundColor: '#01b757',
+        color: '#ffffff',
+        fontSize: '14px',
+        padding: '12px',
+        border: 'none',
+        borderRadius: '4px',
+        cursor: 'pointer',
+        width: '200px',
+        margin: '10px 0',
+    };
+
+    const downloadArrayAsFile = () => {
+        const data = JSON.stringify(array, null, 2);
+        const blob = new Blob([data], { type: 'application/json' });
+
+        // Temporary anchor element.
+        const a = document.createElement('a');
+        a.href = URL.createObjectURL(blob);
+        a.download = filename;
+        document.body.appendChild(a);
+
+        // Programmatically trigger a click event on the anchor element.
+        a.click();
+
+        // Remove the anchor element from the document body.
+        document.body.removeChild(a);
+
+        URL.revokeObjectURL(a.href);
+    };
+
+    return (
+        <button type="button" style={buttonStyle} onClick={downloadArrayAsFile}>
+            Download Logs
+        </button>
+    );
+};
+
+const useLogWorker = (setLogs: React.Dispatch<React.SetStateAction<any[]>>) => {
+    const logWorker = new SharedWorker('./workers/shared-logger-worker.js');
+    useEffect(() => {
+        logWorker.port.onmessage = function (event) {
+            console.log('event', event);
+            const { data } = event;
+            switch (data.type) {
+                case 'get-logs':
+                    setLogs(data.payload);
+                    break;
+                case 'log-entry':
+                    setLogs(prevLogs => [...prevLogs, data.payload]);
```

Comment:
lets implement some cap. I suggest 10000 records for now. I played a bit with heap snapshots and it looked like this after 1000 transport-error messages.
<img width="202" alt="image" src="https://github.com/trezor/trezor-suite/assets/30367552/b4dd8658-3715-4aef-879e-5ecd72d54744">

this could actually eat out entire memory after some time.

Reply:
Yes, I added the limit in log.tsx so it only display 1000 messages but also in the sharedworker so it will keep only 1000 messages.